File: time_test.py
import Meanders_Beta.all_functions as mbaf


iterations = 5
for n in range(2, 20, 2):
    local_time = 0
    for i in range(iterations):
        local_meanders = mbaf.Meanders(n)
        local_meanders.get_all_meanders()
        local_time += local_meanders.speed
        print(i, local_meanders.speed)
    print(f"Для меандров размера {n} среднее время работы составляет {local_time / iterations}")
    print()

File: time_test.cpp
#include "Meanders_Cpp/functions.h"
#include <iostream>

int main() {
    int iterations, start, finish;
    std::cout << "Введите через пробел 3 числа: первый и последний класс меандров для проверки времени и ";
    std::cout << "количество итераций для каждого класса" << std::endl;
    std::cin >> start >> finish >> iterations;
    for (int n = start; n < finish + 1; n += 2) {
        double local_time = 0.0;
        for (int i = 0; i < iterations; ++i) {
            Meanders local_meanders(n, true);
            local_time += local_meanders.speed();
            std::cout << i << " " << local_meanders.speed() << std::endl;
        }
        std::cout << "Для меандров размера " << n;
        std::cout << " среднее время работы составляет " << local_time / iterations << std::endl;
        std::cout << std::endl;
    }

    return 0;
}
